### PR TITLE
Support dapp call native function

### PR DESCRIPTION
**Pull request by cezres**

### CODE REVIEW

File: Neuron/Sections/Dapp/Native/DAppGyroscopeMessageHandler.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+class DAppGyroscopeMessageHandler: DAppNativeMessageHandler {
+    struct Parameters: Decodable {
+        let interval: Interval
+    }
+    enum Interval: String, Decodable {
+        case game
+        case ui
+        case normal
+    }
+    enum MessageName: String {
```
Comment (ashchan) [inline Neuron/Sections/Dapp/Native/DAppGyroscopeMessageHandler.swift]: Add a blank line between enum and  types.

File: Neuron/Sections/Dapp/Native/DAppNativeMessageHandler.swift [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        addNativeFunctionHandler(handler: DAppQRCodeMessageHandler())
+        addNativeFunctionHandler(handler: DAppDeviceMotionMessageHandler())
+        addNativeFunctionHandler(handler: DAppGyroscopeMessageHandler())
+    }
+}
+
+extension UIResponder {
+    var viewController: UIViewController? {
+        if self.isKind(of: UIViewController.self) {
+            return self as? UIViewController
```
Comment (ashchan) [inline Neuron/Sections/Dapp/Native/DAppNativeMessageHandler.swift]: Will simply check if let vc = self as? UIViewController work?

Comment (cezres) [reply Neuron/Sections/Dapp/Native/DAppNativeMessageHandler.swift]: @ashchan   这里在执行 `self as? UIViewController` 之前，已经通过了 `self.isKind(of: UIViewController.self)` 的判断

Comment (ashchan) [reply Neuron/Sections/Dapp/Native/DAppNativeMessageHandler.swift]: What I meant is it’s unnecessary to check twice. Also in Swift there is a ‘is’ operator. (I think isKind is ObjC)

Comment (cezres) [reply Neuron/Sections/Dapp/Native/DAppNativeMessageHandler.swift]: 哦！ 对！！！    习惯性的用了 `isKind`

**Pull request merged**